pkg/tasks/metricsserver.go

```diff
@@ -104,9 +105,18 @@ func (t *MetricsServerTask) create(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("reconciling MetricsServer Service failed: %w", err)
 		}
-	}
-	{
-		dep, err := t.factory.MetricsServerDeployment()
+
+		kubeletCABundleConfigMap, err := t.client.WaitForConfigMapByNsName(
+			ctx,
+			types.NamespacedName{
+				Namespace: s.Namespace,
+				Name:      "kubelet-serving-ca-bundle",
```

Comment:
maybe use `PrometheusK8sKubeletServingCABundle` + `WaitForConfigMap` to avoid hardcoding.